ci/nightly/coverage.py

```diff
@@ -19,48 +19,71 @@
 pipeline.template.yml and the docstring on `trim_pipeline` below.
 """
 
+import subprocess
 import sys
 from pathlib import Path
 
-import yaml
-
 import materialize.cli.mzcompose
 
 
 def main() -> int:
-    with open(Path(__file__).parent.parent / "test" / "pipeline.template.yml") as f:
-        pipeline = yaml.safe_load(f)
-
-    tests = []
-
-    for step in pipeline["steps"]:
-        for plugin in step.get("plugins", []):
-            for plugin_name, plugin_config in plugin.items():
-                if plugin_name == "./ci/plugins/mzcompose":
-                    tests.append((plugin_config["composition"], plugin_config["run"]))
+    # Just the fast sqllogictests, for now
+    tests = [("sqllogictest", "default")]
 
     for (composition, workflow) in tests:
-        print(f"==> Running workflow {workflow} in {composition}")
         materialize.cli.mzcompose.main(
             [
-                "run",
-                workflow,
-                "--coverage",
                 "--find",
                 composition,
+                "--coverage",
+                "run",
+                workflow,
             ]
         )
         materialize.cli.mzcompose.main(
             [
-                "down",
-                "-v",
-                "--coverage",
                 "--find",
                 composition,
+                "down",
+                "-v",
             ]
         )
 
-    # TODO: gather and combine coverage information.
+    # NB: mzcompose _munge_services() sets LLVM_PROFILE_FILE
+    subprocess.run(
+        [
+            "rust-profdata",
+            "merge",
+            "-sparse",
+            *Path("test/sqllogictest/coverage").glob("*.profraw"),
```

Comment:
Are we sure each profraw file has a unique name? environmentd and clusterd might be colliding with same name.

Reply:
Yes, that's the problem with the environment variable set in mzcompose; FYI, reviewing this right now is probably not very useful since I am working on it.  Or you could take it over as it is if you want?

Reply:
Nono, just wrote down my thoughts as I read it out of interest, not a real review.